docs/DeveloperGuide.adoc

```diff
@@ -333,6 +333,69 @@ _{Explain here how the data encryption feature will be implemented}_
 
 // end::dataencryption[]
 
+=== Return a Loan feature
+==== Proposed Implementation
+
+Returning a loan is done by setting the corresponding status of the loan to returned.
+
+The steps that have to be done by LoanBook is as follows:
+* Store the current time as the return time of the loan
+* Change the enum of `LoanStatus` to become `Status.RETURNED`
+* Calculate the cost of the loan and display it in the gui as a suggestion for the user.
+
+These steps would change the given loan object, and update the properties of the loan object. The appropriate values stored within the Loan object would therefore change accordingly.
+
+Given below is an example usage scenerio and how the internals of the Loan would behave:
+
+**Step 1.** The user would have to do a search for which loan his would like to return. This will pull up a list of Loans from which the user would be able to select the correct loan.
+
+[NOTE]
+Users who are able to remember the loanID would be able to access the loan directly by using the `i/<LOANID>` tag in the command. This would allow them to skip this step, and directly reference the loan which they would like to modify.
+
+image::return-loan-list-of-loans.png[]
+
+The user lists all loans. This is displayed on the GUI.
+
+{nbsp} +
+
+**Step 2.** The user decides to return the current loan (in the case of the above image, the user wants to return loan number 3, as selected). User will therefore key in `return i/3`.
+
+The LoanBook will save the current time into the `Optional<LoanTime> returnTime` field. This would be done by calling the constructor `LoanTime()`. Also, the loanStatus field would also be updated from `ONGOING` to `RETURNED`
+
+image::return-loan-loan2.png[]
+Note that originally the loan has a `null` value for end time. Note that `endTime` and `loanStatus` values has been updated.
+
+{nbsp} +
+
+**Step 3.** The function now calculates the cost of the loan. This price would be based on the amount of time the loan was active for, as well as the loanRate that was set, by multiplying the time with the rate.
+
+[NOTE]
+Although the time saved is to the second, the time that is multiplied when we are getting the cost is rounded down to the nearest minute. In a sense, this is "pro-rating" the cost, and making it more discrete.
+
+The result is then displayed into the GUI for the user as the cost price of the loan.
+
+[WARNING]
+The loan book is set up so that if the `endTime` is before `startTime`, the cost returned is 0!
```

Comment:
Perhaps you could mention what mechanism(s) is/are in place to prevent this scenario from occurring.

Reply:
It is set up that way in the code. Must I reference the checks in the code?

Reply:
As in it's clearly an error to have the endTime occur before the startTime. What is the justification for this scenario being allowed, or should there be mechanisms (e.g. exception throwing) when this is detected?

Reply:
OK. We will change the specs of the LoanBook to disallow such a scenario.